src/views/settings/Msg/MsgTemplate/index.vue

```diff
@@ -0,0 +1,150 @@
+              this.showHelpDialog = true
+            }
+            return (
+              <i onClick={handleClick} class='fa fa-question-circle' style='cursor: pointer'>{this.$t('Help')}</i>
+            )
+          },
+          component: Select2,
+          el: {
+            multiple: false,
+            options: []
+          },
+          on: {
+            input: ([event], updateForm) => {
+              setTimeout(() => {
+                vm.templates.map(item => {
+                  if (item.template_name === event) {
+                    this.selectTemplateName = item.template_name
+                    localStorage.setItem('selectTemplateName', item.template_name)
+                    this.variables = item.contexts
+                    this.source = item.source
+                    updateForm({
+                      template_content: item.content.trimStart()
+                    })
+                  }
+                })
+              },
+              500
+              )
+            }
+          }
+        },
+        template_content: {
+          component: MarkDownEditor,
+          on: {
+            htmlChange: ([html]) => {
+              vm.html = html
+            }
+          }
+        }
+      },
+      templates: [],
+      successUrl: { name: 'Msg' },
+      showHelpDialog: false,
+      variables: [],
+      html: '',
+      source: 'original',
+      selectTemplateName: '',
+      variablesHelpText: this.$t('TemplateVariablesHelpText'),
+      hasSaveContinue: false,
+      onPerformError() {
+      },
+      performSubmit(validValues) {
+        validValues['render_html'] = vm.html
+        return this.$axios['patch']('/api/v1/notifications/templates/edit/', validValues).then(res => {
+          this.$router.push({ name: 'Msg', query: { t: new Date().getTime() } })
+        })
+      },
+      moreButtons: [
+        {
+          title: this.$t('Reset'),
+          type: 'default',
+          // hidden: () => this.source === 'original',
+          callback: (value, form, btn) => {
+            return this.$axios['post']('/api/v1/notifications/templates/reset/', { template_name: this.selectTemplateName }).then(
+              () => {
+                this.$router.push({ name: 'Msg', query: { t: new Date().getTime() } })
+                this.$message.success(this.$t('ResetSuccessfully'))
+              }
+            )
+          }
+        }
+      ]
+    }
+  },
+  computed: {
+    variable() {
+      return variable
+    }
+  },
+  mounted() {
+    this.fetchTemplates()
+  },
+  methods: {
+    fetchTemplates() {
+      this.$axios.get('/api/v1/notifications/templates/').then(data => {
+        if (data.length > 0) {
+          this.templates = data
+          this.fieldsMeta.template_name.el.options = data.map(item => ({
+            label: item.subject,
+            value: item.template_name
+          }))
+        }
+      })
+    }
+  }
+}
+</script>
+
+<style scoped>
+
+</style>
```

Comment:
There is no code provided to be checked. Please provide the code you want to have analyzed and reviewed.